pkg/storage/cacher_test.go

```diff
@@ -339,6 +339,32 @@ func TestWatch(t *testing.T) {
 	close(fakeClient.WatchResponse)
 }
 
+func TestWatcherTimeout(t *testing.T) {
+	server, etcdStorage := newEtcdTestStorage(t, testapi.Default.Codec(), etcdtest.PathPrefix())
```

Comment:
I just realized, that it may not be pachable, because this is using "in-memory etcd" which I think is not cherrypicked to 1.1 branch.
If that's true, I think that we should just cherrypick the code change without the test - WDYT?